pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -667,19 +701,13 @@ func (r *ReconcileClusterDeployment) reconcileCompletedProvision(cd *hivev1.Clus
 		statusChange = true
 		cd.Status.Conditions = conds
 	}
-	if provision.Spec.AdminKubeconfigSecretRef != nil && (cd.Status.WebConsoleURL == "" || cd.Status.APIURL == "") {
+	if cd.Spec.ClusterMetadata != nil &&
+		cd.Spec.ClusterMetadata.AdminKubeconfigSecretRef.Name != "" &&
+		(cd.Status.WebConsoleURL == "" || cd.Status.APIURL == "") {
+
 		statusChange = true
-		adminKubeconfigSecret := &corev1.Secret{}
-		if err := r.Get(context.Background(), types.NamespacedName{Namespace: cd.Namespace, Name: provision.Spec.AdminKubeconfigSecretRef.Name}, adminKubeconfigSecret); err != nil {
-			cdLog.WithError(err).Error("failed to get admin kubeconfig secret")
-			return reconcile.Result{}, err
-		}
-		if err := r.fixupAdminKubeconfigSecret(adminKubeconfigSecret, cdLog); err != nil {
-			cdLog.WithError(err).Error("failed to fix up admin kubeconfig secret")
-			return reconcile.Result{}, err
-		}
-		if err := r.setAdminKubeconfigStatus(cd, cdLog); err != nil {
-			cdLog.WithError(err).Error("failed to set admin kubeconfig status")
+		err := r.postProcessAdminKubeconfig(cd, cdLog)
```

Comment:
The controller will end up updating the status of the cluster deployment twice--once in this function call and once a few lines later. I suggest that we move the status update out of the `postProcessAdminKubeconfig` method.

Reply:
Updated.